### ISSUE TITLE
[@typescript-eslint/padding-line-between-statements] False negative in namespace declaration block

### ISSUE THREAD

**Issue by josephzidell** [typescript-eslint/typescript-eslint#3863]
- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/padding-line-between-statements": ["error", {
        {blankLine: 'always', prev: '*', next: ['interface', 'type']}
    }]
  }
}
```

```TS
declare namespace Types {
	type Foo = string;
	type Bar = string;
	interface FooBar {
		[key: string]: string;
	}
	interface BarBaz {
		[key: string]: string;
	}
}
```

**Expected Result**

Expected blank line before this statement.

**Actual Result**

No errors/warnings shown

**Additional Info**

Outside of a namespace declaration block, missing a blank line does report an error for each of these. But inside a namespace declaration block, no missing padding lines are reported as errors.

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `4.31.0` |
| `@typescript-eslint/parser`        | `4.31.0` |
| `TypeScript`                       | `4.4.2` |
| `ESLint`                           | `7.32.0` |
| `node`                             | `v14.16.1` |

**Comment by josephzidell** [comment]
Thanks @bradzacher!

**Comment by josephzidell** [comment]
@rafaelss95, this solution definitely works to force newlines between statements within a `declare` block. However, it seems to also force a newline before the first item.

To wit, even the test scenario produces an error. In fact, I wonder why there are 4 expected errors when there should have been 3?

Code:
```ts
declare namespace Types {
	type Foo = string;

	type Bar = string;

	interface FooBar {
		[key: string]: string;
	}

	interface BarBaz {
		[key: string]: string;
	}
}
```

Output:
```
error: Expected blank line before this statement (@typescript-eslint/padding-line-between-statements) at src/types.d.ts:4:2:
  2 | 
  3 | declare namespace Types {
> 4 |   type Foo = string;
    |   ^
  5 | 
  6 |   type Bar = string;
  7 | 
```
